homeassistant/components/switchbee/climate.py

```diff
@@ -0,0 +1,189 @@
+    """Representation of a SwitchBee climate."""
+
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
+    )
+    _attr_fan_modes = SUPPORTED_FAN_MODES
+    _attr_target_temperature_step = 1
+
+    def __init__(
+        self,
+        device: SwitchBeeThermostat,
+        coordinator: SwitchBeeCoordinator,
+    ) -> None:
+        """Initialize the Switchbee switch."""
+        super().__init__(device, coordinator)
+        # set HVAC capabilities
+        self._attr_max_temp = device.max_temperature
+        self._attr_min_temp = device.min_temperature
+        self._attr_temperature_unit = HVAC_UNIT_SB_TO_HASS[device.unit]
+        self._attr_hvac_modes = [HVAC_MODE_SB_TO_HASS[mode] for mode in device.modes]
+        self._attr_hvac_modes.append(HVACMode.OFF)
+        self._update_attrs_from_coordinator()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_attrs_from_coordinator()
+        super()._handle_coordinator_update()
+
+    def _update_attrs_from_coordinator(self) -> None:
+
+        coordinator_device = self._get_coordinator_device()
+
+        self._attr_hvac_mode: HVACMode = (
+            HVACMode.OFF
+            if coordinator_device.state == ApiStateCommand.OFF
+            else HVAC_MODE_SB_TO_HASS[coordinator_device.mode]
+        )
+        self._attr_fan_mode = FAN_SB_TO_HASS[coordinator_device.fan]
+        self._attr_current_temperature = coordinator_device.temperature
+        self._attr_target_temperature = coordinator_device.target_temperature
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set hvac mode."""
+
+        if hvac_mode == HVACMode.OFF:
+            await self._operate(power=ApiStateCommand.OFF)
+        else:
+            await self._operate(
+                power=ApiStateCommand.ON, mode=HVAC_MODE_HASS_TO_SB[hvac_mode]
+            )
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        await self._operate(target_temperature=kwargs[ATTR_TEMPERATURE])
+
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
+        """Set AC fan mode."""
+        await self._operate(fan=FAN_HASS_TO_SB[fan_mode])
+
+    async def _operate(
+        self,
+        power: str | None = None,
+        mode: str | None = None,
+        fan: str | None = None,
+        target_temperature: int | None = None,
+    ) -> None:
+        """Send request to central unit."""
+
+        assert isinstance(self.hvac_mode, HVACMode)
+        assert isinstance(self.fan_mode, str)
+        assert isinstance(self.target_temperature, int)
+
+        new_power = power
+        if not new_power:
+            new_power = (
+                ApiStateCommand.OFF
+                if self.hvac_mode == HVACMode.OFF
+                else ApiStateCommand.ON
+            )
```

Comment:
```suggestion
            if self.hvac_mode == HVACMode.OFF:
                new_power = ApiStateCommand.OFF
            else:
                new_power = ApiStateCommand.ON
```
It is recommended to avoid using ternary operator when it spans over multiple lines

Another short option is to assign a default and update it:
```python
            new_power = ApiStateCommand.ON
            if self.hvac_mode == HVACMode.OFF:
                new_power = ApiStateCommand.OFF
```